src/Components/Facility/LiveFeedTile.tsx

```diff
@@ -0,0 +1,1045 @@
+//           <button
+//             className="rounded-md border border-white bg-green-200 px-3 py-2 font-semibold text-black hover:bg-green-300"
+//             onClick={() => {
+//               setShowDefaultPresets(!showDefaultPresets);
+//             }}
+//           >
+//             {showDefaultPresets
+//               ? t("show_patient_presets")
+//               : t("show_default_presets")}
+//           </button>
+//           {showDefaultPresets
+//             ? viewOptions.map((option: any) => (
+//                 <div
+//                   onClick={() => {
+//                     setLoading(true);
+//                     gotoPreset(option.value);
+//                   }}
+//                 >
+//                   <button className="w-full rounded-md border border-white bg-green-100 px-3 py-2 font-semibold text-black hover:bg-green-200">
+//                     {option.label}
+//                   </button>
+//                 </div>
+//               ))
+//             : bedPresets.map((preset: any, index: number) => (
+//                 <div
+//                   onClick={() => {
+//                     setLoading(true);
+//                     gotoBedPreset(preset);
+//                   }}
+//                   key={preset.id}
+//                 >
+//                   <button className="w-full rounded-md border border-white bg-green-100 px-3 py-2 font-semibold text-black hover:bg-green-200">
+//                     {preset.meta.preset_name
+//                       ? preset.meta.preset_name
+//                       : `Unnamed Preset ${index + 1}`}
+//                   </button>
+//                 </div>
+//               ))}
+//         </div>
+//       </div>
+//     </div>
+//   );
+// }
+
+import { useEffect, useState, useRef } from "react";
+import { useDispatch } from "react-redux";
+import useKeyboardShortcut from "use-keyboard-shortcut";
+import {
+  listAssetBeds,
+  partialUpdateAssetBed,
+  deleteAssetBed,
+} from "../../Redux/actions";
+import { getCameraPTZ } from "../../Common/constants";
+import {
+  StreamStatus,
+  useMSEMediaPlayer,
+} from "../../Common/hooks/useMSEplayer";
+import { useFeedPTZ } from "../../Common/hooks/useFeedPTZ";
+import * as Notification from "../../Utils/Notifications.js";
+import { AxiosError } from "axios";
+import { BedSelect } from "../Common/BedSelect";
+import { BedModel } from "./models";
+import useWindowDimensions from "../../Common/hooks/useWindowDimensions";
+import CareIcon from "../../CAREUI/icons/CareIcon";
+import ConfirmDialog from "../Common/ConfirmDialog";
+import { FieldLabel } from "../Form/FormFields/FormField";
+import useFullscreen from "../../Common/hooks/useFullscreen";
+import { FeedCameraPTZHelpButton } from "./Consultations/Feed";
+
+const LiveFeed = (props: any) => {
+  const middlewareHostname =
+    props.middlewareHostname || "dev_middleware.coronasafe.live";
```

Comment:
We seem to be duplicating this a lot. This logic seems hard to maintain across multiple places.